### ISSUE TITLE
[BUG] Commit7190799ac75ba78f7f2b21860a884a1d2891952e Introduces Breaking Change

### ISSUE THREAD

**Issue by ColeMurray** [cython/cython#3793]
**Describe the bug**
When utilizing Cython 0.29.21, which includes commit 7190799ac75ba78f7f2b21860a884a1d2891952e, I am experiencing the following stack
```
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Build/Dependencies.py", line 1110, in cythonize
      cythonize_one(*args)
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Build/Dependencies.py", line 1256, in cythonize_one
      result = compile_single(pyx_file, options, full_module_name=full_module_name)
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Compiler/Main.py", line 575, in compile_single
      return run_pipeline(source, options, full_module_name)
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Compiler/Main.py", line 494, in run_pipeline
      pipeline = Pipeline.create_pyx_pipeline(context, options, result)
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Compiler/Pipeline.py", line 252, in create_pyx_pipeline
      create_pipeline(context, mode, exclude_classes=exclude_classes),
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Compiler/Pipeline.py", line 144, in create_pipeline

�
[tox]
      from .ParseTreeTransforms import WithTransform, NormalizeTree, PostParse, PxdPostParse
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Compiler/ParseTreeTransforms.py", line 1632, in <module>
      class AnalyseDeclarationsTransform(EnvTransform):
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Compiler/ParseTreeTransforms.py", line 1640, in AnalyseDeclarationsTransform
      """, level='c_class', pipeline=[NormalizeTree(None)])
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Compiler/TreeFragment.py", line 235, in __init__
      mod = t = parse_from_strings(name, fmt_code, fmt_pxds, level=level, initial_pos=initial_pos)
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Compiler/TreeFragment.py", line 75, in parse_from_strings
      scope = scope, context = context, initial_pos = initial_pos)
    File "Cython/Compiler/Scanning.py", line 306, in Cython.Compiler.Scanning.PyrexScanner.__init__
    File "Cython/Compiler/Scanning.py", line 35, in Cython.Compiler.Scanning.get_lexicon
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Compiler/Lexicon.py", line 147, in make_lexicon
      (Eof, Method('eof_action'))
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Plex/Lexicons.py", line 129, in __init__
      nfa, default_initial_state, spec, token_number)
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Plex/Lexicons.py", line 162, in add_token_to_machine
      match_bol=1, nocase=0)
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Plex/Regexps.py", line 301, in build_machine
      re.build_machine(m, s1, s2, match_bol, nocase)
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Plex/Regexps.py", line 340, in build_machine
      initial_state = self.build_opt(m, initial_state, BOL)
    File "/tmp/pip-build-env-jkbqfqc0/overlay/lib/python3.5/site-packages/Cython/Plex/Regexps.py", line 136, in build_opt
      initial_state.link_to(s)
    File "Cython/Plex/Machines.py", line 86, in Cython.Plex.Machines.Node.link_to
    File "Cython/Plex/Machines.py", line 82, in Cython.Plex.Machines.Node.add_transition
    File "Cython/Plex/Transitions.py", line 60, in Cython.Plex.Transitions.TransitionMap.add
    File "Cython/Plex/Machines.py", line 124, in Cython.Plex.Machines.Node.__hash__
  OverflowError: Python int too large to convert to C ssize_t
```
The error is introduced with the addition of the ```__hash__``` function here: https://github.com/cython/cython/blame/4d54aeff34753551cf0ac9977d50c292dbf9d5d5/Cython/Plex/Machines.py#L124

**To Reproduce**
Code to reproduce the behaviour:
```
pip install gevent
```
The issue is occurring when building gevent, but gevent is not the root cause. 

**Expected behavior**
Expect code to compile correctly

**Environment (please complete the following information):**
 - OS: i386/debian:stretch (docker)
 - Python version [e.g. 3.5.2]
 - Cython version [e.g. 0.29.21]

I am utilizing docker & venv. A psuedo script to repro
```
docker pull i386/debian:stretch
```
*Dockerfile*
```
FROM i386/debian:stretch

RUN apt-get update -y --fix-missing
RUN apt-get install python3.5-dev
RUN python3 -m venv venv --without-pip
RUN . venv/bin/activate && curl https://bootstrap.pypa.io/get-pip.py | python &&  pip install gevent
```

**Additional context**
I believe the issue is due to the i386 architecture with  ssize_t being 32-bit causing the overflow.

**Comment by da-woods** [comment]
It definitely isn't 0.29.2, since that predates the change by > 1 year. I don't think that commit went into 0.29.x at all - it only went into the 3 alpha branch. So I'm not at all clear on why version you're using.

I think (but I'm not 100% sure) that this was fixed in https://github.com/cython/cython/commit/ef423d1e9c9e67a24db0e61b7b334983858fa581. Again, only in the 3 alpha branch

**Comment by ColeMurray** [comment]
> It definitely isn't 0.29.2, since that predates the change by > 1 year. I don't think that commit went into 0.29.x at all - it only went into the 3 alpha branch. So I'm not at all clear on why version you're using.
> 
> I think (but I'm not 100% sure) that this was fixed in [ef423d1](https://github.com/cython/cython/commit/ef423d1e9c9e67a24db0e61b7b334983858fa581). Again, only in the 3 alpha branch

Fixed typo. 0.29.21*

**Comment by da-woods** [comment]
Closing as a duplicate of https://github.com/cython/cython/issues/3659
